src/codeflare_sdk/utils/generate_yaml.py

```diff
@@ -219,14 +219,22 @@ def update_rayclient_ingress(
     spec["rules"][0]["host"] = f"rayclient-{cluster_name}-{namespace}.{ingress_domain}"
 
 
-def update_names(yaml, item, appwrapper_name, cluster_name, namespace):
+def update_names(yaml, item, appwrapper_name, cluster_name, namespace, openshift_oauth):
     metadata = yaml.get("metadata")
     metadata["name"] = appwrapper_name
     metadata["namespace"] = namespace
     lower_meta = item.get("generictemplate", {}).get("metadata")
     lower_meta["labels"]["workload.codeflare.dev/appwrapper"] = appwrapper_name
+    lower_meta["annotations"]["codeflare.dev/oauth"] = f"{openshift_oauth}"
     lower_meta["name"] = cluster_name
     lower_meta["namespace"] = namespace
+    lower_spec = item.get("generictemplate", {}).get("spec")
+    if openshift_oauth:
+        lower_spec["headGroupSpec"]["template"]["spec"]["containers"][0]["env"][1][
+            "valueFrom"
+        ]["secretKeyRef"]["name"] = f"{cluster_name}-oauth-config"
+    if not openshift_oauth:
+        del lower_spec["headGroupSpec"]["template"]["spec"]["containers"][0]["env"][1]
```

Comment:
Can you change this to:
```python
if openshift_oauth:
    lower_spec["headGroupSpec"]["template"]["spec"]["containers"][0]["env"].append(
        {...}
    )
```
(better yet get the container by name ('ray-head') rather than index.
I'm not a huge fan of how the codebase relies on indexes and editing and deleting values from the template. I think it's a bit too easy to accidentally break.

Reply:
Thanks Kevin, that actually cleaned up the code too and was able to remove the cookie_secret from the base-template. I went for the first approach this time, the second approach could be something to look at too, but find that it involves a series of if-for loops. - Will investigate more on this, perhaps a refactor should be made to the codebase to not rely on indexes.

Reply:
I agree, I've been slowly doing this with new changes but a concerted effort might be worthwhile